pkg/telemetryservice/sql/tdengine/tdengine.go

```diff
@@ -12,23 +12,54 @@ import (
 	_ "github.com/taosdata/driver-go/v3/taosRestful"
 )
 
+const (
+	DeviceNameHeaderField = "device_name"
+	EventTagHeaderField   = "event_tag"
+)
+
 type DBHelper struct {
 	DB       *sql.DB
 	Settings *v1alpha1.SQLConnectionSetting
 }
 
-func SendToTDengine(ctx context.Context, rawData []byte, sqlcs *v1alpha1.SQLConnectionSetting) error {
+type DeviceInfo struct {
+	Name string // device name
+	Tag  string // device tag
+}
+
+type StructureData struct {
+	DeviceName string    `json:"devicename"` // device name
+	Timestamp  time.Time `json:"ts"`         // event time stamp
+	Data       string    `json:"data"`       // event content
+	Tag        string    `json:"tg"`         // event categories
+}
+
+func SendToTDengine(ctx context.Context, rawData []byte, sqlcs *v1alpha1.SQLConnectionSetting, header map[string][]string) error {
 	db := &DBHelper{Settings: sqlcs}
 
 	err := db.connectToTDengine(ctx)
 	if err != nil {
-		logger.Errorf("Error to Connect to tdengine, error %v", err.Error())
+		logger.Errorf("Error to Connect to tdengine, error %s", err.Error())
 		return err
 	}
 
-	err = db.insertDataToDB(ctx, rawData)
+	deviceInfo := &DeviceInfo{}
+	if name, exists := header[DeviceNameHeaderField]; exists {
+		deviceInfo.Name = name[0]
+	} else {
+		logger.Infof("Error to get device name from http header")
```

Comment:
if this is an error should use logger.Errorf,
else just logger.Infof("empty device name from http header")